tools/driver/p4c_src/main.py

```diff
@@ -136,14 +136,18 @@ def main():
                         "description (default is binary). "
                         "[Deprecated; use '--p4runtime-files' instead]",
                         action="store", default="binary")
-    parser.add_argument("--target-help", dest="show_target_help",
+    parser.add_argument("--pragmas-help", "--help-pragmas", "--help-pragma",
+                        dest="help_pragmas", action="store_true", default=False,
+                        help = "Print the documentation about supported pragmas and exit.")
```

Comment:
P4₁₆ officially calls these annotations, not pragmas. Neither annotations nor pragmas appear in the P4₁₄ spec.

Reply:
Even though it is not in the spec, it is in the grammar -- this is how we write annotations in P4-14:
```
v1lexer.ll:68:"@pragma"[ \t]*[A-Za-z_][A-Za-z0-9_]* {
v1lexer.ll:71:"@pragma"[ \t]* { BEGIN((driver.saveState = PRAGMA_LINE));
v1lexer.ll:72:                  return Parser::make_PRAGMA("pragma", driver.yylloc); }
v1parser.ypp:229:                                    field_or_masked_ref opt_condition pragma_operand
v1parser.ypp:232:                                    field_match_list expressions pragma_operands
v1parser.ypp:271:    | input PRAGMA pragma_operands NEWLINE
v1parser.ypp:957:pragma_operands: /* epsilon */
v1parser.ypp:959:    | pragma_operands pragma_operand
v1parser.ypp:961:    | pragma_operands "," pragma_operand
```

Reply:
I guess I am advocating for consistency in what these things are called. Granted, `--annotation-help` seems relatively unwieldy. Also, if we are keeping with "pragma" over "annotation", what about adding `--pragma-help` to the list of variants?

Reply:
Having seen your most recent changes, these seem the most natural to me: `--annotation-help`, `--help-annotations`, `--pragma-help`, `--help-pragmas`. Of course, it doesn't hurt to have all eight(!) variants.

Reply:
I like this. Also, it helps that args_parse will actually handle `--help-pragma` if it resolves uniquely, and it does resolve to `--help-pragmas`. So we only need 6 variants ...